setup.py

```diff
@@ -10,7 +10,7 @@
     author_email='vinicius.chiele@gmail.com',
     description='Flask-IO is a library for parsing Flask request arguments into parameters and for serialization of complex objects into Flask response.',
     keywords=['flask', 'rest', 'parse', 'encode', 'decode', 'request', 'json', 'marshmallow'],
-    install_requires=['flask>=0.10.1', 'python-dateutil>=2.4.2', 'marshmallow>=2.9.1,<3'],
+    install_requires=['flask>=0.10.1,<=1.1.4', 'python-dateutil>=2.4.2', 'marshmallow>=3.0.0,<=3.14.1'],
```

Comment:
Pinning to any version of 3 might be better here
`marshmallow>=3`

Reply:
I think we can use `marshmallow>=3,<4`, the same for flask `flask>=0.10.1,<3`. What do you think?

Reply:
Ye we can do that. I noticed that flask v2+ does not support any python version <=3.5 now though and this project is listed as having a max python version of 3.4
Python 3.6 is EOL now though so libraries not supporting any version less than that is something to be expected. 
If people want to use this library on those versions they can pin to v1.14.3

Reply:
Ok, so let's also not support python version <= 3.5 and force the dependencies to `flask>=2,<3` and `marshmallow>=3,<4`. Do you mind making those changes in this PR?
Once it is merged I will release a `v2`

Reply:
Ye sure, I'll do that now. Do you have a good way to test all these changes? I've done some regression and the unittests are passing but there are a lot of scenarios I haven't covered yet.

Reply:
Ye I've done some more testing and its just some marshmallow syntax that needs to be updated on my end. 
The unit tests seem to cover the main functionality of the library already so I'm not sure where I could improve the coverage. Functionality wise there should be no change to flask-io